core/object/object.cpp

```diff
@@ -64,7 +64,7 @@ struct _ObjectDebugLock {
 
 #endif
 
-STRUCT_LAYOUT(PropertyInfoLayout, "PropertyInfo", 5,
+STRUCT_LAYOUT(PropertyInfoLayout, "PropertyInfo",
```

Comment:
This should go in the .h I guess.

Reply:
Putting it in the .h makes things messy because then object.h has to include struct.h (where STRUCT_LAYOUT is defined) and now most of the engine suddenly depends on struct.h. Keeping this part in object.cpp keeps struct.h much more isolated.

Reply:
Hmm. Well I just now realized that putting the definition of `Struct<PropertyInfoLayout>` in object.cpp means that anyone that wants to use `Struct<PropertyInfoLayout>` has to include object.cpp, which is not good. Maybe the STRUCT_LAYOUT macro should go somewhere more central than struct.h? What would make sense? type_defs.h?

Reply:
Ah, no, I think the solution is to just put it in object.h or make another filed in the variant folder called struct_layout.h. I was getting issues with circular dependencies if I tried to put the definition in typed_defs.h or type_info.h.